cpp/src/parquet/stream_writer.h

```diff
@@ -0,0 +1,181 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef PARQUET_STREAM_WRITER_H
+#define PARQUET_STREAM_WRITER_H
+
+#include <array>
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "arrow/util/string_view.h"
+#include "parquet/api/writer.h"
+
+namespace parquet {
+
+// A class for writing Parquet files using an output stream type API.
+//
+// The values given must be of the correct type i.e. the type must
+// match the file schema exactly otherwise a ParquetException will be
+// thrown.
+//
+// The user must explicitly indicate the end of the row using the
+// EndRow() function or EndRow output manipulator.
+//
+// A maximum row group size can be configured, the default size is
+// 512MB.  Alternatively the row group size can be set to zero and the
+// user can create new row groups by calling the EndRowGroup()
+// function or using the EndRowGroup output manipulator.
+//
+// Currently there is no support for optional or repeated fields.
+//
+class PARQUET_EXPORT StreamWriter {
+ public:
+  StreamWriter() = default;
```

Comment:
is the object in a valid state if constructed with a default constructor?

Reply:
Please see reply given for same comment on StreamReader class.